File: backend/src/routes/categories.ts
import { Router } from 'express';
import { Recipe } from '../models/Recipe';

const router = Router();

router.get('/categories', async (req, res) => {
  try {
    // Fetch all unique categories from existing recipes
    const recipes = await Recipe.find({}, 'categories');
    const uniqueCategories = [...new Set(recipes.flatMap(recipe => recipe.categories || []))];
    
    res.json(uniqueCategories);
  } catch (error) {
    console.error('Error fetching categories:', error);
    res.status(500).json({ error: 'Failed to fetch categories' });
  }
});

export default router; 

File: backend/ourRecipesBack/routes/categories.py
from flask import jsonify
from flask_jwt_extended import jwt_required
from ..models.recipe import Recipe
from flask import Blueprint

categories_bp = Blueprint('categories', __name__)

@categories_bp.route('', methods=['GET'])
@jwt_required()
def get_categories():
    """Get all unique categories from recipes"""
    try:
        categories_set = set()
        recipes = Recipe.query.all()
        
        for recipe in recipes:
            if recipe._categories:
                categories_set.update(recipe.categories)
        
        return jsonify({"data": sorted(list(categories_set))}), 200

    except Exception as e:
        print(f"Error fetching categories: {str(e)}", flush=True)
        return jsonify({"error": "Failed to fetch categories"}), 500 